mwdb/web/src/commons/api/index.js

```diff
@@ -362,6 +362,74 @@ function getConfigStats(fromTime) {
     });
 }
 
+function getRemoteObject(remote, type, id) {
+    return axios.get(`/remote/${remote}/api/${type}/${id}`);
+}
+
+function getRemoteObjectList(remote, type, older_than, query) {
+    return axios.get(`/remote/${remote}/api/${type}`, {
+        params: { older_than, query },
+    });
+}
+
+function getRemoteObjectCount(remote, type, query) {
+    return axios.get(`/remote/${remote}/api/${type}/count`, {
+        params: { query },
+    });
+}
+
+function getRemoteObjectTags(remote, id) {
+    return axios.get(`/remote/${remote}/api/object/${id}/tag`);
+}
+
+function getRemoteObjectComments(remote, id) {
+    return axios.get(`/remote/${remote}/api/object/${id}/comment`);
+}
+
+function getRemoteObjectRelations(remote, id) {
+    return axios.get(`/remote/${remote}/api/object/${id}/relations`);
+}
+
+function getRemoteObjectShares(remote, id) {
+    return axios.get(`/remote/${remote}/api/object/${id}/share`);
+}
+
+function getRemoteObjectMetakeys(remote, id) {
+    return axios.get(`/remote/${remote}/api/object/${id}/meta`);
+}
+
+function requestRemoteFileDownload(remote, id) {
+    return axios.post(`/remote/${remote}/api/request/sample/${id}`);
+}
+
+function addRemoteObjectFavorite(remote, id) {
+    return axios.put(`/remote/${remote}/api/object/${id}/favorite`);
+}
+
+function removeRemoteObjectFavorite(remote, id) {
+    return axios.delete(`/remote/${remote}/api/object/${id}/favorite`);
+}
+
+function remoteSearch(remote, query) {
+    return axios.post(`/remote/${remote}/api/search`, { query });
+}
+
+function addRemoteQuickQuery(remote, type, name, query) {
+    return axios.post(`/remote/${remote}/api/${type}/quick_query`, {
+        type,
+        name,
+        query,
+    });
+}
+
+function getRemoteQuickQueries(remote, type) {
+    return axios.get(`/remote/${remote}/api/${type}/quick_query`);
+}
+
+function deleteRemoteQuickQuery(remote, id) {
+    return axios.delete(`/remote/${remote}/api/quick_query/${id}`);
+}
```

Comment:
These will be shared by all local instance users, I'm not sure if "quick queries" should be enabled in remote mode.

Reply:
Ok, I will block this action, but what about favorites? In object view and search? Should it be hidden or displayed?